pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/BranchConditionalScript.groovy

```diff
@@ -36,7 +36,11 @@ class BranchConditionalScript extends DeclarativeStageConditionalScript<BranchCo
 
     @Override
      boolean evaluate() {
-        return describable.branchMatches(describable.compare,
-            (String)script.getProperty("env").getProperty("BRANCH_NAME"))
+        String branchName = (String)script.getProperty("env").getProperty("BRANCH_NAME")
+        if (branchName == null || branchName == "") {
```

Comment:
`StringUtils.isEmpty(branchName)`

Reply:
well actually this is groovy so:
```groovy
 if(!branchName?.trim())
```
maybe? I'm a bit unsure about the `trim()` part :)

Reply:
Or maybe not :) https://gist.github.com/kdabir/2867339